moveit_ros/planning_interface/robot_interface/src/wrap_python_robot_interface.cpp

```diff
@@ -209,7 +236,7 @@ class RobotInterfacePython : protected py_bindings_tools::ROScppInitializer
   {
     if (!current_state_monitor_)
     {
-      ROS_ERROR("Unable to get current robot state");
+      // we do not monitor the current state
```

Comment:
I don't know what I think about the existence of `current_state_monitor_` expressing the intention to monitor the current state. What if the current state is supposed to be monitored but the `current_state_monitor_` was not assigned correctly? I assume the error message in the previous version existed for a reason.

Even if that's safe, a member variable storing `monitor_current_state` as set in the consructor would be easier to read imo.

Reply:
[current_state_monitor_ was previously constructed unconditionally in the constructor](https://github.com/ros-planning/moveit/blob/8f0d7934c35dd07afd93854f694246d4c5fd3417/moveit_ros/planning_interface/robot_interface/src/wrap_python_robot_interface.cpp#L67) and the check was redundant.

It seems quite valid to me to use a `nullptr` value in a `shared_ptr` to check for whether a feature is enabled. :confused:

Reply:
Was it really redundant though, or was that checking for a connection failure or something?
I'm not familiar enough with the monitors, but can that constructor fail somehow so that `current_state_monitor_` is a `nullptr` although the user expressed the intention to monitor the current state?

It's fine with me if this is no problem, just double checking.

Reply:
> Was it really redundant though, or was that checking for a connection failure or something?

I looked through the relevant code again. No, there is no way the pointer could become `nullptr` before.